api/block/routes.go

```diff
@@ -50,161 +46,117 @@ type APIMiniBlock struct {
 		return
 	}
 
-	endpointThrottler, ok := ef.GetThrottlerForEndpoint(endpoint)
-	if ok {
-		if !endpointThrottler.CanProcess() {
-			c.JSON(
-				http.StatusTooManyRequests,
-				shared.GenericAPIResponse{
-					Data:  nil,
-					Error: errors.ErrTooManyRequests.Error(),
-					Code:  shared.ReturnCodeSystemBusy,
-				},
-			)
-			return
-		}
-
-		endpointThrottler.StartProcessing()
-		defer endpointThrottler.EndProcessing()
+	nonce, err := getQueryParamNonce(c)
+	if err != nil {
+		shared.RespondWith(
+			c,
+			http.StatusBadRequest,
+			nil,
+			fmt.Sprintf("%s: %s", errors.ErrValidation.Error(), errors.ErrInvalidBlockNonce.Error()),
+			shared.ReturnCodeRequestError,
+		)
+		return
 	}
 
-	nonceStr := c.Param("nonce")
-	nonce, err := strconv.ParseUint(nonceStr, 10, 64)
-	if nonceStr == "" || err != nil {
-		c.JSON(
-			http.StatusBadRequest,
-			shared.GenericAPIResponse{
-				Data:  nil,
-				Error: fmt.Sprintf("%s: %s", errors.ErrValidation.Error(), errors.ErrInvalidBlockNonce.Error()),
-				Code:  shared.ReturnCodeRequestError,
-			},
+	withTxs, err := getQueryParamWithTxs(c)
+	if err != nil {
+		shared.RespondWith(
+			c,
+			http.StatusInternalServerError,
+			nil,
+			fmt.Sprintf("%s: %s", errors.ErrValidation.Error(), errors.ErrInvalidQueryParameter.Error()),
+			shared.ReturnCodeRequestError,
 		)
 		return
 	}
 
 	block, err := ef.GetBlockByNonce(nonce, withTxs)
 	if err != nil {
-		c.JSON(
+		shared.RespondWith(
+			c,
 			http.StatusInternalServerError,
-			shared.GenericAPIResponse{
-				Data:  nil,
-				Error: errors.ErrGetBlock.Error(),
-				Code:  shared.ReturnCodeInternalError,
-			},
+			nil,
+			errors.ErrGetBlock.Error(),
+			shared.ReturnCodeInternalError,
 		)
 		return
 	}
 
-	c.JSON(
-		http.StatusOK,
-		shared.GenericAPIResponse{
-			Data:  gin.H{"block": block},
-			Error: "",
-			Code:  shared.ReturnCodeSuccess,
-		},
-	)
+	shared.RespondWith(c, http.StatusOK, gin.H{"block": block}, "", shared.ReturnCodeSuccess)
+
 }
 
-func getBlkByHash(c *gin.Context, withTxs bool, endpoint string) {
-	ef, ok := c.MustGet("elrondFacade").(BlkService)
+func getBlockByHash(c *gin.Context) {
+	ef, ok := c.MustGet("elrondFacade").(BlockService)
 	if !ok {
-		c.JSON(
-			http.StatusInternalServerError,
-			shared.GenericAPIResponse{
-				Data:  nil,
-				Error: errors.ErrInvalidAppContext.Error(),
-				Code:  shared.ReturnCodeInternalError,
-			},
-		)
+		shared.RespondWithInvalidAppContext(c)
 		return
 	}
 
-	endpointThrottler, ok := ef.GetThrottlerForEndpoint(endpoint)
-	if ok {
-		if !endpointThrottler.CanProcess() {
-			c.JSON(
-				http.StatusTooManyRequests,
-				shared.GenericAPIResponse{
-					Data:  nil,
-					Error: errors.ErrTooManyRequests.Error(),
-					Code:  shared.ReturnCodeSystemBusy,
-				},
-			)
-			return
-		}
-
-		endpointThrottler.StartProcessing()
-		defer endpointThrottler.EndProcessing()
-	}
-
 	hash := c.Param("hash")
 	if hash == "" {
-		c.JSON(
+		shared.RespondWith(
```

Comment:
We can also add a helper method like `shared.RespondWithValidationError`, but not required.

Reply:
done.